File: cpp/bc2026/src/bestsofar/Utils/AdditionalClasses/Threat.cpp
#include "Utils/AdditionalClasses/Threat.hpp"

namespace Utils {
namespace AdditionalClasses {

// The Java fields are `static` (see Threat.hpp) - one shared instance.
MapLocation Threat::loc;
Direction Threat::dir;
bool Threat::threat = false;

}  // namespace AdditionalClasses
}  // namespace Utils

File: py/bc2026/src/bestsofar/Utils/AdditionalClasses/Threat.py
# Port of bestsofar/Utils/AdditionalClasses/Threat.java. See py/CONVENTIONS.md.
from api import *


class Threat:
    # Java: the fields are declared `static` (arguably a bug - every Threat
    # instance shares one loc/dir/threat). Preserved exactly.
    loc = None
    dir = None
    threat = False

    def __init__(self, m, d, t):
        Threat.loc = m
        Threat.dir = d
        Threat.threat = t
